python/feathub/feature_tables/feature_table.py

```diff
@@ -93,3 +93,15 @@ def get_feature(self, feature_name: str) -> Feature:
             transform=feature_name,
             keys=self.keys,
         )
+
+    def is_bounded(self) -> bool:
+        # FeatureTable is bounded by default unless subclass overwrite the method.
+        return True
+
+    def get_bounded_view(self) -> "FeatureTable":
```

Comment:
Is there any specific benefit of having this method return `FeatureTable` instead of `TableDescriptor`?

If not, it seems simpler and more consistent with the abstract method definition to just return `TableDescriptor` here?

Same for other `get_bounded_view()` overrides.

Reply:
I think it is common in Java to return covariant type when overriding methods. Although not obvious, the benefit is that the user/developer can get a more specific return type when calling the method in the subclass, which can avoid type casting. [1](https://www.whizlabs.com/blog/covariant-return-types-in-java/)[2](https://www.section.io/engineering-education/method-overriding-and-overloading-using-covariant-return-type-in-java/). What do you think? 

Currently, in unit tests, we can avoid typecasting.

Reply:
Hmm.. can you explain which test may fail (due to lack of typecasting) if we update all `get_bounded_view()` overrides to return `TableDescriptor`?

Covariant type is useful if the subclass has a public method not found in the parent class which we want to invoke after calling `get_bounded_view()`. Do you have a concrete use-case here?

Reply:
It is line 107 in `test_kafka_utils.py`, where calling the `get_table_from_source` requires a `FeatureTable` but `get_bounded_view` returns `TableDescriptor`. 

I think the subclass has no public method that is not found in the parent class. And due to the lack of a use case here. You are right that we should prefer consistency here. I have updated the return type of the `get_bounded_view()` overrides.